src/foam/dao/AbstractF3FileJournal.js

```diff
@@ -0,0 +1,485 @@
+        'java.io.IOException'
+      ],
+      args: [
+        {
+          name: 'x',
+          type: 'Context'
+        },
+        {
+          name: 'record',
+          type: 'String'
+        },
+        {
+          name: 'c',
+          type: 'String'
+        },
+        {
+          name: 'prefix',
+          type: 'String'
+        }
+      ],
+      javaCode: `
+        write_(sb.get()
+          .append(prefix)
+          .append("p(")
+          .append(record)
+          .append(")")
+          .append(c)
+          .toString());
+      `
+    },
+    {
+      name: 'remove',
+      type: 'FObject',
+      args: [
+        { name: 'x',      type: 'Context' },
+        { name: 'prefix', type: 'String' },
+        { name: 'dao',    type: 'DAO' },
+        { name: 'obj',    type: 'foam.core.FObject' }
+      ],
+      javaCode: `
+      final Object id = obj.getProperty("id");
+
+      getLine().enqueue(new foam.util.concurrent.AbstractAssembly() {
+        String record_ = null;
+
+        public Object[] requestLocks() {
+          return new Object[] { id };
+        }
+
+        public void executeUnderLock() {
+          dao.remove_(x, obj);
+        }
+
+        public void executeJob() {
+          try {
+            // TODO: Would be more efficient to output the ID portion of the object.  But
+            // if ID is an alias or multi part id we should only output the
+            // true properties that ID/MultiPartID maps too.
+            FObject toWrite = (FObject) obj.getClassInfo().newInstance();
+            toWrite.setProperty("id", obj.getProperty("id"));
+            record_ = getOutputter().stringify(toWrite);
+          } catch (Throwable t) {
+            getLogger().error("Failed to write put entry to journal", t);
+          }
+        }
+
+        public void endJob() {
+          if ( foam.util.SafetyUtil.isEmpty(record_) ) return;
+
+          try {
+            writeComment_(x, obj);
+            writeRemove_(x, record_, foam.util.SafetyUtil.isEmpty(prefix) ? "" : prefix + ".");
+
+            if ( isLast() ) getWriter().flush();
+          } catch (Throwable t) {
+            getLogger().error("Failed to write put entry to journal", t);
+          }
+        }
+      });
+
+      return obj;
+      `
+    },
+    {
+      name: 'writeRemove_',
+      javaThrows: [
+        'java.io.IOException'
+      ],
+      args: [
+        {
+          name: 'x',
+          type: 'Context'
+        },
+        {
+          name: 'record',
+          type: 'String'
+        },
+        {
+          name: 'prefix',
+          type: 'String'
+        }
+      ],
+      javaCode: `
+        write_(sb.get()
+          .append(prefix)
+          .append("r(")
+          .append(record)
+          .append(")")
+          .toString());
+      `
+    },
+    {
+      name: 'write_',
+     // synchronized: true,
+      javaThrows: [
+        'java.io.IOException'
+      ],
+      args: [
+        {
+          class: 'String',
```

Comment:
Make this be a CharSequence instead of a String. Then you can still specify String but also a StringBuilder which avoids the extra step of converting SB's to Strings.
https://docs.oracle.com/javase/7/docs/api/java/lang/CharSequence.html